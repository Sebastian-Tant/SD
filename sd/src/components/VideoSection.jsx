import React from "react";
//yes
const VideoSection = ({ facilityImage }) => {
  return (
    <section className="video-container">
      {facilityImage ? (
        <secion className="image-wrapper">
          <img
            className="facility-image"
            src={facilityImage}
            alt="Selected Facility"
            onError={(e) => {
              console.error("Image failed to load:", facilityImage);
              e.target.style.display = "none"; // Hide on error
            }}
          />
        </secion>
      ) : (
        <video className="booking-video" autoPlay muted loop data-testid="booking-video">
          <source src="/videos/tennis.mp4" type="video/mp4" />
          Your browser does not support the video tag.
        </video>
      )}
    </section>
  );
};
export default VideoSection;