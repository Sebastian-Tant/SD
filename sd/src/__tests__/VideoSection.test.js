import React from "react"; // ðŸ‘ˆ Required for JSX to work
import { render, screen } from "@testing-library/react";
import VideoSection from "../components/VideoSection";

describe("VideoSection Component", () => {
  test("renders video element with correct attributes", () => {
    render(<VideoSection />);
    const videoElement = screen.getByTestId("booking-video");
    expect(videoElement).toBeInTheDocument();
    expect(videoElement).toHaveClass("booking-video");
    expect(videoElement).toHaveAttribute("autoPlay");
    expect(videoElement).toHaveAttribute("loop");
    expect(videoElement.muted).toBe(true);
});

  test("contains source element with correct video file", () => {
    render(<VideoSection />);
    const videoElement = screen.getByTestId("booking-video");
    const sourceElement = videoElement.querySelector("source");
    expect(sourceElement).toBeInTheDocument();
    expect(sourceElement).toHaveAttribute("src", "/videos/tennis.mp4");
    expect(sourceElement).toHaveAttribute("type", "video/mp4");
  });

  test("shows fallback text for unsupported browsers", () => {
    render(<VideoSection />);
    expect(screen.getByText("Your browser does not support the video tag.")).toBeInTheDocument();
  });

});
