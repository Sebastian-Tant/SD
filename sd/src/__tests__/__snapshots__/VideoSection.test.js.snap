// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoSection Component matches snapshot 1`] = `
<DocumentFragment>
  <video
    autoplay=""
    class="booking-video"
    data-testid="booking-video"
    loop=""
  >
    <source
      src="/videos/tennis.mp4"
      type="video/mp4"
    />
    Your browser does not support the video tag.
  </video>
</DocumentFragment>
`;
